book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/admin/service/AdminService.java

```diff
@@ -0,0 +1,89 @@
+import org.lbcc.bms.bms_monolith.common.response.ApiResponse;
+import org.lbcc.bms.bms_monolith.admin.exceptions.VendorNotFoundException;
+import org.lbcc.bms.bms_monolith.admin.repository.VendorRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class AdminService {
+
+    private final VendorRepository vendorRepository;
+
+    public AdminService(VendorRepository vendorRepository) {
+        this.vendorRepository = vendorRepository;
+    }
+
+    public VendorOnboardResponse onboardNewVendor(VendorOnboardRequest vendorOnboardRequest) {
+        log.info("Onboarding new vendor {}", vendorOnboardRequest.getName());
+        //TODO: will save the logoFile to file storage like s3 and get the URL and then update the vendor object before saving
+        Vendor vendor = vendorRepository.save(VendorOnboardRequest.buildVendorFromDto(vendorOnboardRequest));
+        log.info("Vendor onboarded successfully with id {}", vendor.getId());
+
+        return new VendorOnboardResponse(vendor.getId().toString(), vendor.getName(), vendor.getStatus());
+    }
+
+    public List<VendorDto> searchVendor(String vendorName) {
+        if (StringUtils.isEmpty(vendorName)) {
+            throw new InvalidRequest("Vendor name cannot be empty");
+        }
+
+        log.info("Searching for vendor with name {}", vendorName);
+        List<Vendor> vendorList = vendorRepository.findByNameContaining(vendorName);
+        log.info("total vendors found {}", vendorList.size());
+        return vendorList.isEmpty() ? List.of() :
+                VendorSearchResponse.buildVendorDtoListFromVendorList(vendorList);
+
+
+    }
+
+    public Vendor findVendorById(UUID vendorId) {
+        if (vendorId == null) {
+            throw new InvalidRequest("Vendor id cannot be empty");
+        }
+        return vendorRepository.findById(vendorId).orElseThrow(() -> new VendorNotFoundException("Vendor not found"));
+    }
+
+    public Vendor updatedVendorStatus(String vendorId, VendorStatus vendorStatus) {
+        validateVendorUpdateRequest(vendorId, vendorStatus);
+        log.info("updating vendor with id {} to {} ", vendorId, vendorStatus);
+        Vendor vendor = findVendorById(UUID.fromString(vendorId));
+        vendor.setStatus(vendorStatus);
+        vendorRepository.save(vendor);
+        log.info("Vendor id {}  suspended successfully", vendorId);
+        return vendor;
+    }
+
+    public Page<Vendor> searchVendors(VendorSearchRequest searchRequest, Pageable pageable) {
+        try {
+            Specification<Vendor> spec = VendorSpecifications.createSpecification(searchRequest);
+            return searchRequest.hasSearchCriteria() ? vendorRepository.findAll(spec, pageable)
+                    : vendorRepository.findAll(pageable);
```

Comment:
you can safely call "return vendorRepository.findAll(spec, pageable)"

For that create a "criteriaBuilder.conjunction(); " criteria. Check out the comments in VendorSpecification.